packages/pluggableWidgets/app-events-native/src/AppEvents.tsx

```diff
@@ -18,61 +18,66 @@ export class AppEvents extends Component<Props> {
     private lastOnOnline = 0;
     private lastOnOffline = 0;
     private onLoadTriggered = false;
-    private timeoutHandle?: any;
+    private onTimeoutAction: ScheduledExecution | undefined;
     private unsubscribeNetworkEventListener?: NetInfoSubscription;
 
     async componentDidMount(): Promise<void> {
-        if (this.props.onResumeAction) {
-            AppState.addEventListener("change", this.onAppStateChangeHandler);
-        }
-
-        if (this.props.onTimeoutAction) {
-            const schedule = this.props.timerType === "once" ? setTimeout : setInterval;
-            this.timeoutHandle = schedule(() => executeAction(this.props.onTimeoutAction), this.props.delayTime * 1000);
-        }
-
-        if (this.props.onOnlineAction || this.props.onOfflineAction) {
-            this.isConnected = (await NetInfo.fetch()).isConnected;
-            this.unsubscribeNetworkEventListener = NetInfo.addEventListener(this.onConnectionChangeHandler);
-        }
-
-        if (!this.onLoadTriggered && this.props.onLoadAction?.canExecute) {
-            this.onLoadTriggered = true;
-            executeAction(this.props.onLoadAction);
-        }
+        AppState.addEventListener("change", this.onAppStateChangeHandler);
+        this.registerScheduledExecution();
+        await this.registerNetworkEvents();
+        this.triggerOnLoadEvent();
     }
 
     componentWillUnmount(): void {
-        if (this.props.onUnloadAction && this.props.onUnloadAction.canExecute) {
-            executeAction(this.props.onUnloadAction);
-        }
+        AppState.removeEventListener("change", this.onAppStateChangeHandler);
+        this.onTimeoutAction?.cancel();
+        this.unRegisterNetworkEvents();
+        // TODO: Suggestion Deprecate this behavior.
```

Comment:
I am bit pessimistic about our approach for reducing capabilities. I would also opt for removing this since (as you said) people will thinking `changing pages` will always yield `onUnmount`. 

I guess it will be clarified in the docs. Will ask this to our PO's

Reply:
I really disagree on that behaviour and it can be summarised with a nice clean widget with a simple api on componentDidMount didUnmount.  That people can safely use in the layout while AppEvents is not safe to be used per page; due to pages in the backstack not unmounting.

Reply:
Please don't make it a doc item. :)